spec.md

```diff
@@ -782,25 +817,28 @@ message ValidateVolumeCapabilitiesRequest {
 }
 
 message ValidateVolumeCapabilitiesResponse {
-  message Result {
-    // True if the Plugin supports the specified capabilities for the
-    // given volume. This field is REQUIRED.
-    bool supported = 1;
+  // True if the Plugin supports the specified capabilities for the
+  // given volume. This field is REQUIRED.
+  bool supported = 1;
 
-    // Message to the CO if `supported` above is false. This field is
-    // OPTIONAL.
-    // An empty string is equal to an unspecified field value.
-    string message = 2;
-  }
-
-  // One of the following fields MUST be specified.
-  oneof reply {
-    Result result = 1;
-    Error error = 2;
-  }
+  // Message to the CO if `supported` above is false. This field is
+  // OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string message = 2;
 }
 ```
 
+##### ValidateVolumeCapabilities Errors
+
+If the plugin is unable to complete the ValidateVolumeCapabilities call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+If the conditions defined below are encountered, the plugin MUST return the specified gRPC error code.
+The CO MUST implement the specified error recovery behavior when it encounters the gRPC error code.
+
+Condition | gRPC Code | Description | Recovery Behavior
+| --- | --- | --- | --- |
+| Unsupported `volume_capability` | 0 OK | Indicates that one or more of the specified `volume_capability` options (mount flags, volume type, FS type, etc.) are not supported. `ValidateVolumeCapabilitiesResponse.supported` MUST be `false`. `ValidateVolumeCapabilitiesResponse.message` must detail what volume capabilities are incompatible. | Caller MUST fix the `volume_capability` before retrying. |
```

Comment:
Why "retry" is needed if the response is OK?

Reply:
That's what I'm wondering. I don't think an error code of 0 is appropriate if one of the options are not supported.

Reply:
The `ValidateVolumeCapabilitiesResponse` response has a `supported` bool. My understanding was that even if capabilities were not compatible, for this RPC the code should be ok and the boolean would indicate that. But based on what @akutz mentioned above it sounds like we can have both a non-OK error code and a `ValidateVolumeCapabilitiesResponse` response. So I'll change this to `INVALID_ARGUMENT`